### PR TITLE
feat(edit-ema): Add styling to the contentlet toolbars #27076

### LINKED ISSUE #27076: Edit EMA: Add styling to the contentlet toolbars
### Parent
#26644 

### User Story
As a dotCMS user, I want to have the ability to add content, widgets, and forms to a page, so I can have a more streamlined way of managing my content and UI elements.

> [!IMPORTANT]
> Right now we have the "add" button in the container level, we need to move that to the contentlet top and botto, because that will give us the position within the content type to add the new content.

> [!NOTE]
> All the styling should be pure native CSS. Because want to avoid css colission we need to try to move this to the shadow dom, here is a library we can try: https://www.npmjs.com/package/react-shadow.

> [!NOTE]
> Because I want to keep this as native as possible to show the popover menu (add button options), we can try to use the popover API: https://caniuse.com/?search=popover

```[tasklist]
### Acceptance Criteria
- [ ] User should be able to add content top or bottom of another contentlet
- [ ] User should be able to add content, forms and widgets
- [ ] The buttons for code and drag is out of this scope
```

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
- [Prototype](https://www.figma.com/proto/trbwjLuf5F6KaU1rFucNeg/Edit-Page?page-id=1902%3A6704&type=design&node-id=1909-28315&viewport=-3068%2C392%2C0.74&t=sw2h0huHLCJCUb5D-1&scaling=scale-down&starting-point-node-id=1909%3A28315&mode=design)
- Figma: [One](https://www.figma.com/file/trbwjLuf5F6KaU1rFucNeg/Edit-Page?type=design&node-id=2341%3A9115&mode=design&t=OcEBZIlGrr65j4xf-1), [Two](https://www.figma.com/file/trbwjLuf5F6KaU1rFucNeg/Edit-Page?type=design&node-id=2286%3A9249&mode=design&t=OcEBZIlGrr65j4xf-1)

Contentlet

### Assumptions & Initiation Needs
We already have an action in place that will add a contentlet in a specific position of a contentlet, here is the interface:

```typescript
export interface ActionPayload {
    container: ContainerPayload;
    contentlet?: ContentletPayload;
    language_id: string;
    pageContainers: PageContainer[];
    pageId: string;
    personaTag?: string;
    position?: 'before' | 'after';
    newContentletId?: string;
}
```

More: `core-web/libs/portlets/dot-ema/src/lib/shared/models.ts`

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates:
- Design the dropdown options and popover: 2 hours
- Implement the dropdown

### PR DESCRIPTION
**Pull request by fmontes**
### Proposed Changes
* Add edit component
* Clean up the react

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/store/dot-ema.store.ts [hunk 1]
```diff
@@ -199,11 +203,34 @@ export class EditEmaStore extends ComponentStore<EditEmaState> {
...
+    //     return {
+    //         ...state,
+    //         dialogVisible: true,
+    //         dialogHeader: 'Search Forms', // Does this need translation?
+    //         dialogIframeLoading: true,
+    //         dialogIframeURL: null,
+    //         dialogType: 'widget'
+    //     };
+    // });
+
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/store/dot-ema.store.ts:233]: Should we remove this and let the code as noops in the editor?

File: core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+.bounds {
+    position: absolute;
+    pointer-events: none;
+    outline: solid 2px $color-palette-primary-500;
+}
+
+.actions {
+    display: flex;
+    gap: 4px;
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.scss:35]: use the variable

File: core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+})
+export class EmaContentletToolsComponent {
+    private buttonPosition: 'after' | 'before' = 'after';
+
+    @Input() contentlet: ContentletArea;
+    @Output() addContent = new EventEmitter<ActionPayload>();
+    @Output() addForm = new EventEmitter<ActionPayload>();
+    @Output() addWidget = new EventEmitter<ActionPayload>();
+    @Output() edit = new EventEmitter<ActionPayload>();
+    @Output() delete = new EventEmitter<ActionPayload>();
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:27]: This is all emitting the same. Should we have one event and send a type to identify the action?

File: ema/nextjs/src/components/dotcms-page.js [hunk 4]
```diff
@@ -56,16 +58,29 @@ export const DotcmsPage = () => {
...
+    const eventScrollHandler = useCallback(
+        (event) => {
+            window.parent.postMessage(
+                {
+                    action: 'scroll'
+                },
+                '*'
+            );
+        },
+        [router]
```
Comment (zJaaal) [inline ema/nextjs/src/components/dotcms-page.js:73]: We don't need the router in the dependencies array

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts (hunk 3 above)
Comment (fmontes) [reply core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:27]: I tried that first but it sucked, harder to test, bigger handlers etc...

File: ema/nextjs/src/components/dotcms-page.js (hunk 4 above)
Comment (fmontes) [reply ema/nextjs/src/components/dotcms-page.js:73]: You added that :D

**Review by zJaaal — APPROVED** [review]

File: ema/nextjs/src/components/dotcms-page.js (hunk 4 above)
Comment (zJaaal) [reply ema/nextjs/src/components/dotcms-page.js]: Oh

**Review by manuelrojas — APPROVED** [review]

**Pull request merged**